x/launch/keeper/msg_server_request_add_account.go

```diff
@@ -16,17 +16,21 @@ func (k msgServer) RequestAddAccount(
 
 	chain, found := k.GetChain(ctx, msg.ChainID)
 	if !found {
-		return nil, sdkerrors.Wrapf(types.ErrChainNotFound, "%v", msg.ChainID)
+		return nil, sdkerrors.Wrapf(types.ErrChainNotFound, "%d", msg.ChainID)
+	}
+
+	if chain.IsMainnet {
+		return nil, sdkerrors.Wrapf(types.ErrChainIsMainnet, "%d", msg.ChainID)
```

Comment:
We can maybe precise in error messages that mainnet can't have account in `launch` or create a specific error type for those